docs/Targets.md

```diff
@@ -0,0 +1,341 @@
+- `Exists`
+- `Push`
+- `Resolve`
+- `Tag`
+- `Predecessors`
+
+## Content Stores
+
+In `oras-go` v2, a content store is an implementation of the [`Target`](#target) interface—specifically, the [`GraphTarget`](#graphtarget) interface.
+
+The library provides four built-in content stores:
+
+- [Memory Store](#memory-store): Stores everything in memory.
+- [OCI Store](#oci-store): Stores content in the OCI-Image layout on the file system.
+- [File Store](#file-store): Stores location-addressable content on the file system.
+- [Repository Store](#repository-store): Communicates with remote artifact repositories (e.g. `ghcr.io`, `docker.io`).
+
+### Memory Store
+
+The memory store, available in the [`content/memory`](https://pkg.go.dev/oras.land/oras-go/v2/content/memory) package, stores all content in memory, where each blob's content is mapped to its corresponding descriptor.
+
+> [!TIP]
+> The memory store is often used for building and storing artifacts in memory before copying them to other stores, such as remote repositories.
+
+### OCI Store
+
+The OCI store, available in the [`content/oci`](https://pkg.go.dev/oras.land/oras-go/v2/content/oci) package, follows the [`OCI image-spec v1.1.1`](https://github.com/opencontainers/image-spec/blob/v1.1.1/image-layout.md) to store blob content on the file system. 
+
+For example, consider an artifact and its signature represented by the following graph:
+
+```mermaid
+graph TD;
+
+SignatureManifest["Signature Manifest<br>(sha256:e5727b...)"]--subject-->Manifest
+SignatureManifest--config-->Config
+SignatureManifest--layers-->SignatureBlob["Signature blob<br>(sha256:37f884)"]
+
+Manifest["Manifest<br>(sha256:314c7f...)"]--config-->Config["Config blob<br>(sha256:44136f...)"]
+Manifest--layers-->Layer0["Layer blob 0<br>(sha256:b5bb9d...)"]
+Manifest--layers-->Layer1["Layer blob 1<br>(sha256:7d865e...)"]
+```
+
+The corresponding directory structure on the file system would look like this:
+
+```bash
+$ tree repo
+repo/
+├── blobs
+│   └── sha256
+│       ├── 314c7f20dd44ee1cca06af399a67f7c463a9f586830d630802d9e365933da9fb
+│       ├── 37f88486592fd90ace303ee38f8d1ff698193e76c76d3c1fef8627a39e677696
+│       ├── 44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a
+│       ├── 7d865e959b2466918c9863afca942d0fb89d7c9ac0c99bafc3749504ded97730
+│       ├── b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c
+│       └── e5727bebbcbbd9996446c34622ca96af67a54219edd58d261112f1af06e2537c
+├── index.json
+├── ingest
+└── oci-layout
+```
+
+In the layout:
+
+- All content, whether manifests or layer blobs, are all placed under the `blobs` directory. The path to each piece of content is determined by its digest.
+- The `index.json` file is an Image Index JSON object. It serves as the entry point for the graph and provides tagging functionality.
+- The `ingest` directory is used temporarily during blob processing. This directory is ORAS-specific and is not defined in the OCI specification.
```

Comment:
Also, this isn't really ORAS-specific. It is also used by `containerd` content store. Precisely, `io.containerd.content.v1.content`.

Reply:
Updated